packages/contracts/contracts/foundry/Sphinx.sol

```diff
@@ -184,44 +191,33 @@ abstract contract Sphinx {
         // Gnosis Safe ensures that its nonce is treated like a contract instead of an EOA.
         sphinxUtils.deploySphinxSystem(systemContracts);
 
-        string memory serializedDeploymentInfo = sphinxCollect(
+        FoundryDeploymentInfo memory deploymentInfo = sphinxCollect(
             _executionMode,
             deployer,
-            _scriptFunctionCalldata
+            _scriptFunctionCalldata,
+            2
+        );
+        vm.writeFile(
+            _deploymentInfoPath,
+            sphinxUtils.serializeFoundryDeploymentInfo(deploymentInfo)
         );
-        vm.writeFile(_deploymentInfoPath, serializedDeploymentInfo);
     }
 
     function sphinxCollect(
         ExecutionMode _executionMode,
         address _executor,
-        bytes memory _scriptFunctionCalldata
-    ) private returns (string memory) {
+        bytes memory _scriptFunctionCalldata,
+        uint64 _callDepth
+    ) private returns (FoundryDeploymentInfo memory) {
         address safe = safeAddress();
         address module = sphinxModule();
 
-        FoundryDeploymentInfo memory deploymentInfo;
-        deploymentInfo.executionMode = _executionMode;
-        deploymentInfo.executorAddress = _executor;
-        deploymentInfo.safeAddress = safe;
-        deploymentInfo.moduleAddress = module;
-        deploymentInfo.chainId = block.chainid;
-        deploymentInfo.blockGasLimit = block.gaslimit;
-        deploymentInfo.safeInitData = sphinxUtils.getGnosisSafeInitializerData(address(this));
-        deploymentInfo.newConfig = SphinxConfig({
-            projectName: sphinxConfig.projectName,
-            owners: sphinxConfig.owners,
-            threshold: sphinxConfig.threshold,
-            orgId: sphinxConfig.orgId,
-            mainnets: sphinxConfig.mainnets,
-            testnets: sphinxConfig.testnets,
-            saltNonce: sphinxConfig.saltNonce
-        });
-        deploymentInfo.initialState = sphinxUtils.getInitialChainState(safe, ISphinxModule(module));
-        deploymentInfo.nonce = sphinxUtils.getMerkleRootNonce(ISphinxModule(module));
-        deploymentInfo.sphinxLibraryVersion = sphinxUtils.getSphinxLibraryVersion();
-        deploymentInfo.arbitraryChain = false;
-        deploymentInfo.requireSuccess = true;
```

Comment:
Moved all of this to `SphinxUtils.sol:initializeDeploymentInfo` to prevent the `Solc.spec.ts` test suite in the demo package from failing due to a "stack too deep" error